code/modules/multiz/movement.dm

```diff
@@ -36,10 +36,19 @@
 		return eyeobj.zMove(direction)
 
 	// Check if we can actually travel a Z-level.
-	if (!can_ztravel())
+	if (!can_ztravel(direction))
```

Comment:
extended this to propagate desired direction through the checks. Important since pipes only go one way